### PR TITLE
Fix RAFT WAL repair.

### PR DESCRIPTION
**Pull request by derekcollison**
When we stored a message in the raft layer in a wrong position (state corrupt), we would panic, leaving the message there.
On restart we would truncate the WAL and try to repair, but we truncated to the wrong index of the bad entry.

This change also includes additional changes to truncateWAL and also reduces the conditional for panic on storeMsg.

Signed-off-by: Derek Collison <derek@nats.io>

/cc @nats-io/core

### CODE REVIEW

**Review by wallyqs — APPROVED** [review]
LGTM

File: server/raft.go [hunk 1]
```diff
@@ -418,12 +420,17 @@ func (s *Server) startRaftNode(cfg *RaftConfig) (RaftNode, error) {
 		for index := state.FirstSeq; index <= state.LastSeq; index++ {
 			ae, err := n.loadEntry(index)
 			if err != nil {
-				n.warn("Could not load %d from WAL [%+v] with error: %v", index, state, err)
-				continue
+				n.warn("Could not load %d from WAL [%+v]: %v", index, state, err)
```
Comment (ripienaar) [inline server/raft.go:423]: These are pretty scary warnings to users, can we add some indication about what they can do about them? Or mention if the server will recover on its own? 

Seeing just these in the server log and no indication of what action to take is not a good experience.  For myself even, I have no idea what to do about these log lines

**Review by wallyqs — APPROVED** [review]
LGTM

File: server/raft.go (hunk 1 above)
Comment (ripienaar) [reply server/raft.go:423]: OK, but like this log says "ask Derek" :) if that's the intended outcome, LGTM :)

**Review by kozlovic — APPROVED** [review]
LGTM

**Pull request merged**